frontend/src/metabase/query_builder/components/GuiQueryEditor.jsx

```diff
@@ -150,7 +157,7 @@ export default class GuiQueryEditor extends React.Component {
         <div className="mx2">
           <PopoverWithTrigger
             id="FilterPopover"
-            ref="filterPopover"
+            ref={this.filterPopover}
             triggerElement={addFilterButton}
             triggerClasses="flex align-center"
             getTarget={() => this.refs.addFilterTarget}
```

Comment:
I have no idea where does this ref come from!? o.O
`addFilterTarget` is mentioned in 2 places only and neither one of them is ref definition:
1. [line 142](https://github.com/metabase/metabase/blob/6b6df6d73660f3297a66c10ab7a2a5149bd67410/frontend/src/metabase/query_builder/components/GuiQueryEditor.jsx#L142)
2. [line 150](https://github.com/metabase/metabase/blob/6b6df6d73660f3297a66c10ab7a2a5149bd67410/frontend/src/metabase/query_builder/components/GuiQueryEditor.jsx#L150)

As I can see, `getTarget` is also not used anywhere else in this file.

Reply:
Seems like `ref={this.filterPopover}` isn't used, so I'd delete and see what happens.

`Popover` and `PopoverWithTrigger` can be passed a target element for `Tether` to use to pin the popover els to (for example, various tooltips in the app are given a ref to a button or icon). The code here (`getTarget`) is wrong--there is no `getTarget` prop and the correct prop to use is `target`, so this code is doing nothing. Maybe worth renaming `getTarget` to `target` to see what happens, but if this code works alright already then I think it is safe to delete as well.

Reply:
I think I'll just delete it. Regardless of non-existing `getTarget` prop, `this.refs.addFilterTarget` also doesn't exist anywhere. That is, `ref="addFilterTarget"` wasn't set.